tools/autograd/load_derivatives.py

```diff
@@ -328,12 +337,20 @@ def check_grad_usage(defn_name: str, derivatives: Sequence[Derivative]) -> None:
                                "otherwise, there is a likely error in your derivatives "
                                "declaration.")
 
+        if used_named_grad and (used_grad or used_grads):
+            raise RuntimeError(
```

Comment:
This error seems reasonable - we'd want to make sure the documentation in derivatives.yaml reflects it though

Reply:
A related but distinct question:

If there is only one differentiable named output. Is it ok to refer to it as `grad_X`, or should it exclusively be referred to by `grad`?